plutus-tx-plugin/test/AssocList/Spec.hs

```diff
@@ -0,0 +1,424 @@
+{-# LANGUAGE BangPatterns          #-}
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NegativeLiterals      #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PatternSynonyms       #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE ViewPatterns          #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:context-level=0 #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:defer-errors #-}
+{-# LANGUAGE FlexibleInstances     #-}
+
+module AssocList.Spec where
+
+import Test.Tasty.Extras
+
+import Data.List (nubBy, sort)
+import Data.Map.Strict qualified as Map
+import Data.Maybe (fromJust)
+import Hedgehog (Gen, MonadTest, Property, Range, forAll, property, (===))
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Range qualified as Range
+import PlutusTx.AssocMap qualified as AssocMap
+import PlutusTx.Builtins qualified as PlutusTx
+import PlutusTx.Code
+import PlutusTx.Data.AssocList (AssocList)
+import PlutusTx.Data.AssocList qualified as Data.AssocList
+import PlutusTx.IsData ()
+import PlutusTx.IsData qualified as P
+import PlutusTx.Lift (liftCodeDef)
+import PlutusTx.Prelude qualified as PlutusTx
+import PlutusTx.Show qualified as PlutusTx
+import PlutusTx.Test
+import PlutusTx.TH (compile)
+import PlutusTx.These (These (..))
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Hedgehog (testProperty)
+
+goldenTests :: TestNested
+goldenTests =
```

Comment:
@zliu41 when you have time, could you provide some documentation for these tests? They are copied over from your original PR.